dashboards/google-vertex-ai/vertex-ai-agent-engine.json

```diff
@@ -0,0 +1,375 @@
+                  "unitOverride": "ms"
+                }
+              }
+            ],
+            "thresholds": [],
+            "yAxis": {
+              "label": "",
+              "scale": "LINEAR"
+            }
+          }
+        }
+      },
+      {
+        "yPos": 40,
+        "height": 16,
+        "width": 24,
+        "widget": {
+          "title": "Query error ratio",
+          "xyChart": {
+            "chartOptions": {
+              "displayHorizontal": false,
+              "mode": "COLOR"
+            },
+            "dataSets": [
+              {
+                "plotType": "LINE",
+                "targetAxis": "Y1",
+                "timeSeriesQuery": {
+                  "prometheusQuery": "sum by (reasoning_engine_id)(rate(aiplatform_googleapis_com:reasoning_engine_request_count{monitored_resource=\"aiplatform.googleapis.com/ReasoningEngine\",response_code!=\"200\"}[${__interval}]))\n/\nsum by (reasoning_engine_id)(rate(aiplatform_googleapis_com:reasoning_engine_request_count{monitored_resource=\"aiplatform.googleapis.com/ReasoningEngine\"}[${__interval}]))\n* 100",
+                  "unitOverride": "%"
+                }
+              }
+            ],
+            "thresholds": [],
+            "yAxis": {
+              "scale": "LINEAR"
+            }
+          }
+        }
+      },
+      {
+        "yPos": 56,
+        "height": 4,
+        "width": 48,
+        "widget": {
+          "title": "Agent Engine Resource Allocations",
+          "id": "",
+          "sectionHeader": {
+            "dividerBelow": false,
+            "subtitle": "This section describes the overall resource allocations. Use these metrics to understand your billing charges, which are calculated based on these metrics."
+          }
+        }
+      },
+      {
+        "yPos": 60,
+        "height": 16,
+        "width": 24,
+        "widget": {
+          "title": "Container CPU allocation",
+          "id": "",
+          "xyChart": {
+            "chartOptions": {
+              "displayHorizontal": false,
+              "mode": "COLOR",
+              "showLegend": false
+            },
+            "dataSets": [
+              {
+                "breakdowns": [],
+                "dimensions": [],
+                "legendTemplate": "",
+                "measures": [],
+                "plotType": "LINE",
+                "targetAxis": "Y1",
+                "timeSeriesQuery": {
+                  "outputFullDuration": false,
+                  "timeSeriesQueryLanguage": "fetch aiplatform.googleapis.com/ReasoningEngine\n| metric 'aiplatform.googleapis.com/reasoning_engine/cpu/allocation_time'\n| align rate(1m)\n| every 1m\n| group_by [resource.reasoning_engine_id],\n    [value_allocation_time_aggregate: aggregate(value.allocation_time)]",
```

Comment:
Does this really have to be MQL? Seems like the query builder would work just fine?
![image](https://github.com/user-attachments/assets/7eb450f3-e5e1-447d-97d3-7acf3059907e)

Same applies to the Memory one below. And if you switched over, you probably wouldn't need to specify a unitOverride either.

Reply:
Same rationale with the other comment.